sentry-uitest/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/SentryBenchmarkTest.kt

```diff
@@ -0,0 +1,135 @@
+package io.sentry.uitest.android.benchmark
+
+import android.content.Context
+import androidx.lifecycle.Lifecycle
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.core.app.launchActivity
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.action.ViewActions.swipeUp
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.runner.AndroidJUnitRunner
+import io.sentry.ITransaction
+import io.sentry.Sentry
+import io.sentry.SentryOptions
+import io.sentry.android.core.SentryAndroid
+import io.sentry.uitest.android.benchmark.util.BenchmarkOperation
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+class SentryBenchmarkTest {
+
+    private lateinit var runner: AndroidJUnitRunner
+    private lateinit var context: Context
+
+    @BeforeTest
+    fun setUp() {
+        runner = InstrumentationRegistry.getInstrumentation() as AndroidJUnitRunner
+        context = ApplicationProvider.getApplicationContext()
+        context.cacheDir.deleteRecursively()
+        IdlingRegistry.getInstance().register(BenchmarkActivity.scrollingIdlingResource)
+    }
+
+    @AfterTest
+    fun cleanup() {
+        IdlingRegistry.getInstance().unregister(BenchmarkActivity.scrollingIdlingResource)
+    }
+
+    @Test
+    fun benchmarkSameOperation() {
+
+        // We compare two operation that are the same. We expect the increases to be negligible, as the results
+        // should be very similar.
+        val op1 = BenchmarkOperation(runner, getOperation(runner) { null })
+        val op2 = BenchmarkOperation(runner, getOperation(runner) { null })
+        val comparisonResult = BenchmarkOperation.compare(op1, "Op1", op2, "Op2")
+
+        assertTrue { comparisonResult.durationIncrease >= -1 }
+        assertTrue { comparisonResult.durationIncrease < 1 }
+        assertTrue { comparisonResult.cpuTimeIncrease >= -1 }
+        assertTrue { comparisonResult.cpuTimeIncrease < 1 }
+        // The fps decrease is skipped for the moment, due to approximation:
+        // if an operation runs at 59.49 fps and the other at 59.51, they are considered 59 and 60 fps respectively.
+        // Their difference would be 1 / 60 * 100 = 1.66666%
+        // On slow devices, a difference of 1 fps on an average of 20 fps would account for 5% decrease.
+        // On even slower devices the difference would be even higher. Let's skip for now, as it's not important anyway.
```

Comment:
Didn't we say that the fps is the most important metric? 😜

Reply:
The most important is the dropped frames, not the fps.
The fps can be the same (as they don't count "partial frames") while the dropped frames consider the time a frame took over the maximum threshold for a fluid application (60 fpr = around 17 milliseconds per frame). 
So, rather than just saying "we dropped 1 frame", we say e.g. "we dropped 0.3 frames", in case a frame took 5 milliseconds more than allowed.